### PR TITLE
Add C++ gradient for Cast

### LINKED ISSUE #14081: Feature Request: C++ gradient for Cast
Implement the gradient for Cast in C++ so that it is available for TF_AddGradients.

This is the Python code that I believe would need to be ported:

https://github.com/tensorflow/tensorflow/blob/27767d8e9c1325979cf32ff5b81c10df9006fd57/tensorflow/python/ops/math_grad.py#L1109-L1120

Will be asking @bpiel for guidance if I get stuck!

### PR DESCRIPTION
**Pull request by kbsriram**
- add ops (Ceil, Round and Rint) that return zero gradients
- minor formatting fixes from clang-format

Tested for flakes with:
```
$ bazel test --runs_per_test=100 tensorflow/cc:gradients_math_grad_test
...
//tensorflow/cc:gradients_math_grad_test                                 PASSED in 1.6s
  Stats over 100 runs: max = 1.6s, min = 0.6s, avg = 1.3s, dev = 0.1s
```

Fixes https://github.com/tensorflow/tensorflow/issues/14081

### CODE REVIEW

**Comment by tensorflowbutler** [comment]
Nagging Reviewer @drpngx, @benoitsteiner: You have been added as a reviewer to this pull request. Please add your review or reassign. It has been 14 days with no activity and the `awaiting review` label has been applied.

File: tensorflow/cc/gradients/math_grad.cc [hunk 1]
```diff
@@ -38,7 +38,15 @@ REGISTER_NO_GRADIENT_OP("NotEqual");
 REGISTER_NO_GRADIENT_OP("LogicalAnd");
 REGISTER_NO_GRADIENT_OP("LogicalOr");
 REGISTER_NO_GRADIENT_OP("LogicalNot");
+// These are really zero (almost everywhere) gradients, but we return
+// NoGradient() following the current python implementation.
+// Additional discussion on this topic:
+// https://github.com/tensorflow/tensorflow/issues/897
+// https://github.com/tensorflow/tensorflow/issues/783
 REGISTER_NO_GRADIENT_OP("Floor");
+REGISTER_NO_GRADIENT_OP("Ceil");
```
Comment (drpngx) [inline tensorflow/cc/gradients/math_grad.cc]: Can we have a separate test for these as well?

**Review by drpngx — CHANGES_REQUESTED** [review]
Thanks! The change is much simpler than it looks -- most of it is formatting.

File: tensorflow/cc/gradients/math_grad.cc (hunk 1 above)
Comment (kbsriram) [reply tensorflow/cc/gradients/math_grad.cc]: Sure thing - I've added a few for the ones I added, as well as for "Floor" which was missing.

(Apologies for also rolling in the formatting diffs - looks like some code got in without clang-format, and it was lazier to fix it together with this change.)

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 30 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 45 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 60 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 75 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 90 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 120 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @caisq: It has been 135 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Review by josh11b — APPROVED** [review]

**Comment by rthadur** [comment]
@kbsriram can you please resolve conflicts

**Comment by rthadur** [comment]
I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen!

**Pull request closed without merging**